### ISSUE TITLE
GUI crashes when terminal drop-down is expanded (Gnome Web Browser)

### ISSUE THREAD

**Issue by neutrino2211** [alex-saunders/glicky#25]
Here is a side-by-side comparison between Google chrome and the Gnome Web Browser. Unfortunately, the page crashes before anything is logged to the console.

![ezgif com-video-to-gif(1)](https://user-images.githubusercontent.com/21989345/56524468-55e71d00-6541-11e9-897d-d61fc8df3d53.gif)

This seems to only bother the default gnome browser but we can't be sure, maybe we should test other browsers and make a compatibility list.

**Comment by alex-saunders** [comment]
> and make a compatibility list

Love the idea of this, do you have any ideas about where we could keep such a list? I suppose it would be like an 'ongoing issue' but it would be good to keep it somewhere outside of a Github issues so that we can use issues for fixing the listed compatibility problems.

**Comment by alex-saunders** [comment]
As far as the actual issue here, I'm not sure how I'm going to best debug this as I don't have access to a Linux machine or Gnome web browser.. will think about the best approach..
@neutrino2211 would you be up for doing some debugging on this?

**Comment by neutrino2211** [comment]
> > and make a compatibility list
> 
> Love the idea of this, do you have any ideas about where we could keep such a list? I suppose it would be like an 'ongoing issue' but it would be good to keep it somewhere outside of a Github issues so that we can use issues for fixing the listed compatibility problems.

Yeah, maybe like a branch where we would add supported browsers to the README

**Comment by neutrino2211** [comment]
And for the GUI issue, I will gladly take a look.

**Comment by alex-saunders** [comment]
Hmm yeh I'm thinking maybe a branch where we have a compatibility.md file that gets updated as we tests browsers might be good.. will think a little more on it before committing to anything though.

**Comment by neutrino2211** [comment]
After numerous attempts, there is finally an error log

![Screenshot from 2019-04-24 22 10 00](https://user-images.githubusercontent.com/21989345/56694203-1275e580-66de-11e9-9b80-53e513687205.png)

The browser never logged any errors till today but before that I placed `console.log`s in places where I thought an error would occur and more times than not it crashed right around `XTermRenderer,createTextInstance`

**Comment by alex-saunders** [comment]
Yeesh that does not look like a fun error. Have had you had any luck finding the root cause of this @neutrino2211 ? It looks like the custom renderer is failing somewhere within the react internals? This could make it really tricky to debug (even harder because I can't access the Gnome browser).

Maybe [here](https://github.com/alex-saunders/glicky/blob/master/src/xterm-renderer/reconciler/index.js#L27) would be a good place to start delving around.
